include/ginkgo/core/base/executor.hpp

```diff
@@ -783,6 +783,40 @@ class ExecutorBase : public Executor {
 };
 
 
+/**
+ * Controls whether the DeviceReset function should be called thanks to a
+ * boolean.
```

Comment:
Could you add some comments about the reset only considered in the last destroyed executor (device_id)?
If some executor with reset on but some with reset off, the resetting call only depends on the last one.
Also, no matter how users set these executors, ginkgo does not call reset when some executors (under ginkgo) are alive.

Reply:
Good point.